src/main/java/ai/elimu/web/admin/application_version/ApplicationVersionCreateController.java

```diff
@@ -125,6 +127,9 @@ public String handleSubmit(
           applicationVersion.setLabel(label);
 
           Integer minSdkVersion = Integer.valueOf(apkMeta.getMinSdkVersion());
+          if (minSdkVersion < MIN_SDK_VERSION) {
+            result.reject("TooLow.sdkVersionCode");
```

Comment:
I suggest removing the "Code" part of this name, just to avoid confusion with the `versionCode`.

```suggestion
            result.reject("TooLow.sdkVersion");
```